### PR TITLE
💄 CMS UI | Updating the Events Component Editor

### PR DESCRIPTION
**Pull request by isaaclombardssw**
The two big things here were adding a time picker instead of a number box, and associating major cities to offset values for context (via the moment timezone library).

🤖🤖🤖

This pull request includes significant updates to the `Events` component to improve timezone handling and date formatting, as well as some changes to the JSON data and package dependencies.

### Updates to `Events` Component:

* `components/blocks/Events.template.tsx`: 
  * Replaced manual timezone formatting with `moment-timezone` for better accuracy and added `Datetime` for date-time input. Updated date and time formatting functions. [[1]](diffhunk://#diff-3116cb530a3601f3aa1a7ca7d01ebe92df8fdc8127f565059e932120d48ebbfdL1-R54) [[2]](diffhunk://#diff-3116cb530a3601f3aa1a7ca7d01ebe92df8fdc8127f565059e932120d48ebbfdL63-R102) [[3]](diffhunk://#diff-3116cb530a3601f3aa1a7ca7d01ebe92df8fdc8127f565059e932120d48ebbfdL85-R113) [[4]](diffhunk://#diff-3116cb530a3601f3aa1a7ca7d01ebe92df8fdc8127f565059e932120d48ebbfdL96-R124) [[5]](diffhunk://#diff-3116cb530a3601f3aa1a7ca7d01ebe92df8fdc8127f565059e932120d48ebbfdL117-R146)
  * Added a new utility function to map cities to their respective timezones using `moment-timezone`.

* `components/blocks/Events.tsx`: 
  * Fixed start time parsing to handle both string and numeric formats and updated the calculation of `startDateUTC`. [[1]](diffhunk://#diff-931b2e0a59cf1287054fb75fc23c8c6332cd2d0f41e58df3aa29ca9645ad9dfcR6) [[2]](diffhunk://#diff-931b2e0a59cf1287054fb75fc23c8c6332cd2d0f41e58df3aa29ca9645ad9dfcR53-R67)

### JSON Data Updates:

* [`components/blocks/EventsTimezones.json`](diffhunk://#diff-7bf255b5ca6d07fff5b8556c993edfb9610dbe32cd01aec0b158eecb49dcee21R1-R20): Added a list of major cities and their respective timezones.
* [`content/blocksPages/home.json`](diffhunk://#diff-eebf320493e165b962792e5b488122b1dc9363375b50a64f25b180010caf001bL434-R436): Updated `startTime` fields to use ISO 8601 date strings instead of numeric values. [[1]](diffhunk://#diff-eebf320493e165b962792e5b488122b1dc9363375b50a64f25b180010caf001bL434-R436) [[2]](diffhunk://#diff-eebf320493e165b962792e5b488122b1dc9363375b50a64f25b180010caf001bR448) [[3]](diffhunk://#diff-eebf320493e165b962792e5b488122b1dc9363375b50a64f25b180010caf001bL457-R460)

### Dependency Updates:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R61): Added `moment-timezone` and `react-datetime` to dependencies. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R61) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R74)

### Code Cleanup:

* `pages/[slug].tsx`: Reorganized import statements for better readability. ([pages/[slug].tsxL1-R5](diffhunk://#diff-f3bbbdb92fd1d1da397061576d4275aaadbcdf1e1952d4cbe94b5928478141fcL1-R5))

🤖🤖🤖

<details>
### General Contributing:

- [ ] Have you followed the guidelines in our [Contributing document](https://github.com/tinacms/tina.io/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### All New Content Submissions: (To be confirmed by reviewer)

- [ ] Title is short & specific
- [ ] Headers are logically ordered & consistent
- [ ] Purpose of document is explained in the first paragraph
- [ ] Procedures are tested and work
- [ ] Any technical concepts are explained or linked to
- [ ] Document follows structure from templates
- [ ] All links work
- [ ] The spelling and grammar checker has been run
- [ ] Graphics and images are clear and useful
- [ ] Any prerequisites and next steps are defined.
</details>

Screenshot 2024-10-02 at 10 46 35 am
Screenshot 2024-10-02 at 10 46 23 am

### CODE REVIEW

**Comment by isaaclombardssw** [comment]
Closes #2173.

File: components/blocks/Events.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import Image from 'next/image';
 import Link from 'next/link';
 import React, { Suspense, useEffect, useRef, useState } from 'react';
 import { FaCheck, FaChevronRight } from 'react-icons/fa';
+import { start } from 'repl';
```
Comment (bradystroud) [inline components/blocks/Events.tsx:6]: whats this

**Review by bradystroud — CHANGES_REQUESTED** [review]
As per our conversation, we need to pick a city, not a timezone. Timezones can change. (daylight savigns)

image



e.g.
in June, Bob books NDC London for Jan 15th **9am**, and sets timezone to Londons current time zone (+1:00 GMT)
London changes from +1:00 GMT to +0:00 GMT on last day of Oct
The event will change from 9am to 8am

File: components/blocks/EventsTimezones.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[
```
Comment (joshbermanssw) [inline components/blocks/EventsTimezones.json:1]: Couple questions
1. Is this hard-coded or can we edit via Tina
2. What if we dont have an event in a major city?
3. Should this be in the components/blocks/ directory?

Comment (isaaclombardssw) [reply components/blocks/EventsTimezones.json:1]: This is hard coded – editing it in Tina could be possible, but it's just for reference really not to be relied upon.

As per my chat with Brady I'm planning to add a warning that lets users know that they'll need to account for any daylight savings time changes when selecting a timezone as a temporary fix – and I'll make another PBI for the eventual "correct" solution. 🥱

File: components/blocks/Events.template.tsx [hunk 3]
```diff
@@ -1,34 +1,53 @@
-import type { Template } from 'tinacms'
-import { NumberField, NumberFieldPlugin, NumberInput, TextField, wrapFieldsWithMeta } from 'tinacms'
-import React from 'react'
+import moment from "moment-timezone";
+import React, { useEffect, useState } from 'react';
+import Datetime from 'react-datetime';
+import "react-datetime/css/react-datetime.css";
```
[the remark this answers is not in the document]
Comment (isaaclombardssw) [reply components/blocks/Events.template.tsx:4]: It seems like if I don't import it, the component stylings aren't applied as expected.

It looks janky but it as a part of the usage instructions 🥴 
https://www.npmjs.com/package/react-datetime#usage

File: components/blocks/Events.tsx (hunk 1 above)
Comment (isaaclombardssw) [reply components/blocks/Events.tsx]: Removed.

File: components/blocks/EventsTimezones.json (hunk 2 above)
Comment (isaaclombardssw) [reply components/blocks/EventsTimezones.json:1]: Quick draft PBI – https://github.com/tinacms/tina.io/issues/2422

**Review by bradystroud — APPROVED** [review]
Looks ok to me.
As per our conversation, there are already PBIs in the backlog to improve this so lets merge it as-is

**Pull request merged**